torch/sparse/_triton_ops.py

```diff
@@ -1159,6 +1196,8 @@ def bsr_dense_addmm(
         torch.bfloat16: tl.float32,
         torch.float32: tl.float64,
         torch.float64: tl.float64,
+        torch.int8: tl.int32,
+        torch.int32: tl.int32,
```

Comment:
Is `torch.int32: tl.int32,` actually needed?

Reply:
no, unless we want bsr_dense_addmm to work with int32 as well. But you are right, it is OT for this PR, I'll remove it.

Reply:
ah, sorry, we actually need it as `out` dtype is int32 which is defined in `_int_bsr_dense_addmm`.

Reply:
Hm, based on the naming of the variable I was just worried that there was silent upcast somewhere and the kernel starts issuing a) `tl.dot(int32, int32) -> int32` instead of b) `tl.dot(int8, int8) -> int32`. I'd expect a) to be much, much slower than b) on A100.

Reply:
> Hm, based on the naming of the variable I was just worried that there was silent upcast somewhere and the kernel starts issuing a) `tl.dot(int32, int32) -> int32` instead of b) `tl.dot(int8, int8) -> int32`.

a) does not happen: the variable `dot_out_dtype` defines only the dtype of `tl.dot` result (read: the dtype of `output_acc` in the a kernel), `tl.dot` arguments are still int8 as nothing changes the dtype of `input`, `dense` and bsr `values`.

Reply:
Hm, but then why does this PR need to add two entries here instead of just one?

Reply:
@cpuhrsch Notice that this PR introduces two functions:
1. `bsr_dense_addmm(<int8 inputs>) -> <int8 output>` - this was the original feature that corresponds to the entry `torch.int8: tl.int32`.
2. `_int_bsr_dense_addmm(<int8 inputs>) -> <int32 output>` - this is due to your request to have an analogy to `_int_mm`. This corresponds to entry `torch.int32: tl.int32`.

If the feature 1. is of no interest, the PR can be refactored to only include 2. What do you think?

Reply:
Oh I see. Ok that makes sense. Just add it to the title of the PR.